### PR TITLE
Integration tests migrate delegation

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Bug-fixes migrate delegation from a staking provider to another staking provider
- Integration tests  
## Proposed changes
- Did some bug fixes on the part with the migrate delegation
- Added a new integration tests with chain simulator 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	fmt.Println(sp1AddressBytes, sp2AddressBytes)
+
+	// do delegate to sp1
+	delegateValue := big.NewInt(0).Mul(big.NewInt(500), chainSimulatorIntegrationTests.OneEGLD)
+	txDelegate1 := chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, 0, sp1AddressBytes, delegateValue, "delegate", gasLimitForDelegate)
+	delegate1Tx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txDelegate1, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegate1Tx)
+
+	err = cs.ChangeEpochs(40)
```
Comment (raduchis) [inline integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:96]: I would add a failing test here, to see that moving in the 30 epochs after the initial stake is also not possible.

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	tx = chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, staking.GetNonce(t, cs, delegator), vm.DelegationManagerSCAddress, big.NewInt(0), dataField, 80_000_000)
+	moveDelegationTx, err = cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, moveDelegationTx)
+
+	eventWithError := moveDelegationTx.Logs.Events[0]
+	require.Equal(t, eventWithError.Identifier, core.SignalErrorOperation)
+	require.Equal(t, "cannot migrate to another service provider during cooldown period, wait 30 more epochs", string(eventWithError.Topics[1]))
+
+	// check delegation values for delegator in sp1 and sp2
```
Comment (raduchis) [inline integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:119]: can you please add a test for moving another 100 EGLD from sp1 to sp2. Should it work or not? I'm not sure. 
How about moving 100 EGLD from sp1 to sp3? 
Maybe there are unit tests for these cases...

File: vm/systemSmartContracts/validator.go [hunk 3]
```diff
@@ -2242,9 +2242,9 @@ func (v *validatorSC) moveStakeFromValidatorToValidator(args *vmcommon.ContractC
 		return returnCode
 	}
 
-	validatorA := args.Arguments[0]
-	validatorB := args.Arguments[1]
-	stakeToMove := big.NewInt(0).SetBytes(args.Arguments[2])
+	validatorA := args.Arguments[1]
+	validatorB := args.Arguments[2]
+	stakeToMove := big.NewInt(0).SetBytes(args.Arguments[0])
```
Comment (raduchis) [inline vm/systemSmartContracts/validator.go:2247]: nice catch! 

TestDelegationManagerSystemSC_ChangeStakingProvider seems not to test the happy path correctly if it worked with the arguments misplaced.

Comment (sasurobert) [reply vm/systemSmartContracts/validator.go:2247]: that is strange - as the happy path seems to do what the new code wants ==))

Comment (sasurobert) [reply vm/systemSmartContracts/validator.go:2247]: ah. not strange - as the tests were fixed as well.

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go (hunk 1 above)
Comment (miiu96) [reply integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:96]: this check is done at line 104

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go (hunk 2 above)
Comment (miiu96) [reply integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:119]: This case will work, you will have the cool-down period on the destination 
You will can migrate again delegation from sp2 only after cool-down period

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go (hunk 1 above)
Comment (raduchis) [reply integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:97]: ok... so you can move instantly after the first stake. No migrations are allowed with less than 30 epochs

File: integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go (hunk 2 above)
Comment (raduchis) [reply integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go:136]: ok

**Review by raduchis — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**